packages/navuser/src/react/index.tsx

```diff
@@ -0,0 +1,113 @@
+import Avatar from '@pluralsight/ps-design-system-avatar'
+import Halo from '@pluralsight/ps-design-system-halo'
+import { PropsOf, RefFor } from '@pluralsight/ps-design-system-util'
+import { compose, css } from 'glamor'
+import React, { ReactNode, MouseEventHandler } from 'react'
+
+import stylesheet from '../css'
+
+const styles = {
+  navUser: (props: NavUserProps) => {
+    const clickable: boolean = !!props.href || !!props.onClick
+
+    return compose(
+      css(stylesheet['.psds-navuser']),
+      clickable && css(stylesheet['.psds-navuser--clickable'])
+    )
+  },
+  meta: () => css(stylesheet['.psds-navuser__plan-name']),
+  name: () => css(stylesheet['.psds-navuser__name']),
+  words: () => css(stylesheet['.psds-navuser__words'])
+}
+
+interface BaseNavUserProps {
+  meta?: ReactNode
+  name?: string
+  src?: string
+}
+interface AnchorProps extends BaseNavUserProps, PropsOf<'a'> {
+  href: string
+  onClick?: MouseEventHandler<HTMLAnchorElement>
+}
+interface ButtonProps extends BaseNavUserProps, PropsOf<'button'> {
+  href?: undefined
+  onClick: MouseEventHandler<HTMLButtonElement>
+}
+interface DivProps extends BaseNavUserProps, PropsOf<'div'> {
+  href?: undefined
+  onClick?: MouseEventHandler<HTMLDivElement>
+}
+
+type NavUserElement = HTMLAnchorElement | HTMLButtonElement | HTMLDivElement
+type NavUserProps = AnchorProps | ButtonProps | DivProps
+type NavUserComponent = React.ForwardRefExoticComponent<NavUserProps> & {
+  (props: AnchorProps, ref?: RefFor<'a'>): JSX.Element
+  (props: ButtonProps, ref?: RefFor<'button'>): JSX.Element
+  (props: DivProps, ref?: RefFor<'div'>): JSX.Element
+}
```

Comment:
we could adopt the render prop polymorphic pattern here as well. I would change the component prop to be slightly more complicated but would made the component internals simpler